internal/config/instance.go

```diff
@@ -216,39 +242,74 @@ func (i *Instance) InstallSource() string {
 
 // ReadInConfig reads in config from the config file
 func (i *Instance) ReadInConfig() error {
-	i.rwLock.RLock()
-	defer i.rwLock.RUnlock()
-	// Prepare viper, which is a library that automates configuration
-	// management between files, env vars and the CLI
-	i.viper.SetConfigName(i.Name())
-	i.viper.SetConfigType(i.Type())
-	i.viper.AddConfigPath(i.configDir.Path)
-	i.viper.AddConfigPath(".")
-
-	if err := i.viper.ReadInConfig(); err != nil {
-		return errs.Wrap(err, "Cannot read config.")
+	configData, err := i.configFileData()
+	if err != nil {
+		return err
 	}
+	i.data = configData
 	return nil
 }
 
-// Save saves the config file
-func (i *Instance) Save() error {
-	if i.noSave {
-		return nil
+func (i *Instance) configFileData() (map[string]interface{}, error) {
```

Comment:
Can we name this `readInConfigFile` or something that makes it explicit that this entails IO operations.